pkg/uploadserver/uploadserver.go

```diff
@@ -264,13 +246,16 @@ func (app *uploadServerApp) Run() (*RunResult, error) {
 
 func (app *uploadServerApp) getTLSConfig() (*tls.Config, error) {
 	if app.config.ServerCertFile == "" || app.config.ServerKeyFile == "" {
+		if !app.config.Insecure {
+			return nil, errors.New("invalid TLS config")
+		}
 		return nil, nil
 	}
 
 	//nolint:gosec // False positive: Min version is not known statically
 	config := &tls.Config{
 		CipherSuites: app.config.CryptoConfig.CipherSuites,
-		ClientAuth:   tls.RequireAndVerifyClientCert,
+		ClientAuth:   tls.VerifyClientCertIfGiven,
```

Comment:
trying to understand if there are implications to VerifyClientCertIfGiven. can't a request now get further ahead?
or is `if len(r.TLS.VerifiedChains) == 0 {` guarding that